### PR TITLE
fixed get_markets to be compatible to display

### PR DESCRIPTION
**Pull request by dmytro-shash**

### CODE REVIEW

File: contracts/controller/src/prices.rs [hunk 1]
```diff
@@ -43,26 +54,38 @@ mod tests {
         return (eth_contract, token_address, user_account);
     }
 
-    use super::*;
-
     #[test]
     fn test_add_get_price() {
```
Comment (markblaize) [inline contracts/controller/src/prices.rs:58]: @dmytro-shash please create a separate unit-test that checks that method produce correct output

File: contracts/controller/src/prices.rs [hunk 2]
```diff
@@ -22,17 +23,27 @@ impl Contract {
         // Update & insert operation
         self.prices.insert(&price.asset_id, &price);
     }
+
+    pub fn get_markets(&self) -> HashMap<AccountId, AccountId> {
+        let mut result: HashMap<AccountId, AccountId> = HashMap::new();
+        for (asset, asset_contract) in self.markets.iter() {
```
Comment (markblaize) [inline contracts/controller/src/prices.rs:29]: @dmytro-shash it's considered as bad practice to use hand-written loops.

Please check if it possible to create HashMap from UnorderedMap.iter()
https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#impl-FromIterator%3C(K%2C%20V)%3E

Also HashMap has several ways to create from Iter, from vector etc.

File: contracts/controller/src/prices.rs [hunk 3]
```diff
@@ -22,17 +23,27 @@ impl Contract {
         // Update & insert operation
         self.prices.insert(&price.asset_id, &price);
     }
+
+    pub fn get_markets(&self) -> HashMap<AccountId, AccountId> {
```
Comment (markblaize) [inline contracts/controller/src/prices.rs:27]: @dmytro-shash Isn't `UnorderedMap` has not representation? Do we really need to convert it to hashmap?

File: contracts/controller/src/admin.rs [hunk 4]
```diff
@@ -37,7 +37,7 @@ impl Contract {
     pub fn remove_market(&mut self, key: AccountId) {
         require!(self.is_valid_admin_call(), "This functionality is allowed to be called by admin or contract only");
 
-        require!(self.markets.contains_key(&key), "Asset by this key doesnt exist");
+        require!(self.markets.get(&key).is_some(), "Asset by this key doesnt exist");
```
Comment (markblaize) [inline contracts/controller/src/admin.rs:40]: @dmytro-shash do we need this check?

Comment (markblaize) [inline contracts/controller/src/admin.rs:40]: @dmytro-shash do we need this check?

**Review by markblaize — CHANGES_REQUESTED** [review]

File: contracts/controller/src/prices.rs (hunk 1 above)
Comment (dmytro-shash) [reply contracts/controller/src/prices.rs:58]: splitted into two diff tests as a outcome 
image

File: contracts/controller/src/prices.rs [hunk 5]
```diff
@@ -22,17 +23,22 @@ impl Contract {
         // Update & insert operation
         self.prices.insert(&price.asset_id, &price);
     }
+
+    pub fn get_markets(&self) -> Vec<(AccountId, AccountId)> {
```
Comment (markblaize) [inline contracts/controller/src/prices.rs:27]: @dmytro-shash vector is not a good representation for the Map

**Review by markblaize — CHANGES_REQUESTED** [review]

**Comment by markblaize** [comment]
colsed in scope of https://app.clickup.com/t/258xt2e

**Pull request closed without merging**